### PR TITLE
Fix timeslot parsing

### LINKED ISSUE #79: fireroad.py can't parse schedule for 15.089?
When I run `python3 update.py` on a freshly cloned copy of this repo, I get the following message as part of the output:

```
=== Update fireroad data ===
Can't parse schedule 15.089: '11'
Got 1923 courses
Skipped 4735 courses due to missing schedules
```

### PR DESCRIPTION
**Pull request by psvenk**
Fixes #79. Properly account for timings that begin in the morning and end in the evening by special-casing these rather than treating evening times as morning times. Add assertions and give more detailed error messages.

This causes 15.089 ("W EVE (11-6 PM)") to be parsed correctly (as 11:00 AM to 6:00 PM) but also results in early morning ROTC classes (such as NS.100) throwing errors since we don't support times before 8:00 AM (these classes were previously misidentified as evening classes).

It would also make sense to overhaul the timeslot system so that early morning classes can be supported, but I'll leave that to another pull request.

### CODE REVIEW

File: scrapers/fireroad.py [hunk 1]
```diff
@@ -26,34 +26,43 @@
 URL = "https://fireroad.mit.edu/courses/all?full=true"
 
 
-def parse_timeslot(day, slot):
-    """Parses a timeslot. Example: parse_timeslot("M", "10-11.30") -> [4, 3]
+def parse_timeslot(day, slot, pm):
```
Comment (dtemkin1) [inline scrapers/fireroad.py:29]: could you update the docstring at the top of the file?

Comment (dtemkin1) [inline scrapers/fireroad.py:29]: could you update the docstring at the top of the file?

File: scrapers/fireroad.py [hunk 2]
```diff
@@ -26,34 +26,43 @@
...
     Args:
     * day (str): The day as a string
     * slot (str): The slot as a string
+    * pm (bool): Whether the timeslot is in the evening
 
     Returns:
     * list[int]: The parsed day and timeslot
     """
-    pm, slot = slot.endswith(" PM"), slot.rstrip(" PM")
+    assert pm == slot.endswith(" PM")
```
Comment (dtemkin1) [inline scrapers/fireroad.py:40]: if this throws an error how is it handled?

**Review by dtemkin1 — CHANGES_REQUESTED** [review]

File: scrapers/fireroad.py (hunk 1 above)
Comment (psvenk) [reply scrapers/fireroad.py:29]: Oh, thanks for catching that. Fixed.

File: scrapers/fireroad.py (hunk 2 above)
Comment (psvenk) [reply scrapers/fireroad.py:40]: It gets logged to stdout and that class gets skipped, just like any other error in parsing a class's schedule:
https://github.com/sipb/hydrant/blob/e266b31e4566f3299714bc56fa3e329cebc7b5c6/scrapers/fireroad.py#L230-L235

(This is what happens to 15.089 on `main` and to the early morning classes on `timeslots`.)

File: scrapers/fireroad.py [hunk 3]
```diff
@@ -222,7 +231,7 @@ def get_course_data(courses, course):
         raw_class.update(parse_schedule(course))
     except Exception as e:
         # if we can't parse the schedule, warn
-        print(f"Can't parse schedule {course_code}: {e}")
+        print(f"Can't parse schedule {course_code}: {repr(e)}")
```
[the remark this answers is not in the document]
Comment (psvenk) [reply scrapers/fireroad.py:234]: Oh, thanks; I forgot about that

**Review by dtemkin1 — APPROVED** [review]
lgtm!

**Pull request merged**